Modules/System/Math/src/Math.Codeunit.al

```diff
@@ -288,5 +288,82 @@ codeunit 710 Math
     begin
         exit(DotNetMath.Truncate(decimalValue));
     end;
+
+    /// <summary>
+    /// Calculates the bitwise AND value of two integers
+    /// </summary>
+    /// <param name="IntegerValue1">The first integer for the evaluation</param>
+    /// <param name="IntegerValue2">The second integer for the evaluation</param>
+    /// <returns>The resulting integer of the bitwise AND operaion</returns>
+    procedure BitwiseAnd(IntegerValue1: Integer; IntegerValue2: Integer) Result: Integer
+    var
+        BitPosition: Integer;
+        BitIndex: Integer;
+    begin
+        InitBitwiseValues(BitPosition, Result);
+
+        for BitIndex := to 32 do begin
+            if (IntegerValue1 mod 2 = 1) and (IntegerValue2 mod 2 = 1) then
+                UpdateBitwiseResult(BitPosition, Result);
+            UpdateBitwiseValues(IntegerValue1, IntegerValue2, BitPosition);
+        end;
+    end;
+
+    /// <summary>
+    /// Calculates the bitwise OR value of two integers
+    /// </summary>
+    /// <param name="IntegerValue1">The first integer for the evaluation</param>
+    /// <param name="IntegerValue2">The second integer for the evaluation</param>
+    /// <returns>The resulting integer of the bitwise OR operaion</returns>
+    procedure BitwiseOr(IntegerValue1: Integer; IntegerValue2: Integer) Result: Integer
+    var
+        BitPosition: Integer;        
+        BitIndex: Integer;
+    begin
+        InitBitwiseValues(BitPosition, Result);
+
+        for BitIndex := to 32 do begin
+            if (IntegerValue1 mod 2 = 1) or (IntegerValue2 mod 2 = 1) then
+                UpdateBitwiseResult(BitPosition, Result);
+            UpdateBitwiseValues(IntegerValue1, IntegerValue2, BitPosition);
+        end;
+    end;
+
+    /// <summary>
+    /// Calculates the bitwise exclusive OR (XOR) value of two integers
+    /// </summary>
+    /// <param name="IntegerValue1">The first integer for the evaluation</param>
+    /// <param name="IntegerValue2">The second integer for the evaluation</param>
+    /// <returns>The resulting integer of the bitwise exclusive OR (XOR) operaion</returns>
+    procedure BitwiseXOr(IntegerValue1: Integer; IntegerValue2: Integer) Result: Integer
+    var
+        BitPosition: Integer;        
+        BitIndex: Integer;
+    begin
+        InitBitwiseValues(BitPosition, Result);
```

Comment:
Once the implementation codeunit is created, these functions should call the same function there.

Reply:
I will rewrite using the implementation codeunit pattern, but writing tests for a mathematical operation that cannot otherwise be calculated in Dynamics is going to be tricky. This is probably why there is currently no test for the math codeunit that I can see.

The only way I can reasonably do that is by creating a reasonably large set of integers and their outcomes and see if they match. It's limited, but if you're ok with that, I will use that approach.

Reply:
These math functions are deterministic, so despite the tests going to be rudimentary, that's alright.
I'm not sure why tests weren't written for the others, but my assumption is that they're using the system.math in dotnet or constants, so that should've been tested thoroughly, and we would not need to.